src/codegen/compile.ml

```diff
@@ -1052,8 +1052,9 @@ module GC = struct
     E.call_import env "ic0" "performance_counter"
 
   let register_globals env =
-    (E.add_global64 env "__mutator_instructions" Mutable 0L;
-     E.add_global64 env "__collector_instructions" Mutable 0L)
+    E.add_global64 env "__mutator_instructions" Mutable 0L;
+    E.add_global64 env "__collector_instructions" Mutable 0L;
+    E.add_global32 env "_HP" Mutable 0l
```

Comment:
Maybe we could add a safe guard that this is not used for the incremental GC (as long as this is not supported)?
```suggestion
    if !Flags.gc_strategy <> Flags.Incremental then
      E.add_global32 env "_HP" Mutable 0l
    else
      ()
```